ui/src/views/Provisioners/ViewWorker/index.jsx

```diff
@@ -288,6 +279,26 @@ export default class ViewWorker extends Component {
                     disabled: actionLoading,
                   }}
                 />
+                {worker.providerId !== NULL_PROVIDER && (
+                  <SpeedDialAction
+                    tooltipOpen
+                    requiresAuth
+                    icon={<DeleteIcon />}
+                    tooltipTitle="Terminate Worker"
+                    onClick={() =>
+                      this.handleTerminateDialogActionOpen(
+                        worker.workerId,
+                        worker.workerGroup,
+                        worker.workerPoolId
+                      )
+                    }
+                    FabProps={{
+                      disabled:
+                        ['stopping', 'stopped'].includes(worker.state) ||
```

Comment:
I know it's minor, but maybe this can be defined in one place (`WorkerTable` uses same logic), like new tiny util function